### PR TITLE
Feature(Experiment): Adding Docker Service Kill experiment

### PR DESCRIPTION
**Pull request by ghoshankur1983**
Signed-off-by: root <root@ip-172-31-21-135.ap-south-1.compute.internal>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Checklist**

* [x] Does this PR have a corresponding GitHub issue?
* [x] Have you included relevant README for the chaoslib/experiment with details?
* [x] Have you added debug messages where necessary? 
* [x] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: experiments/generic/docker_service_kill/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Experiment Metadata
+
+<table>
+<tr>
+<th> Name </th>
+<th> Description </th>
+<th> Documentation Link </th>
+</tr>
+<tr>
+ <td> Kubelet Service Kill </td>
```
Comment (ispeakc0de) [inline experiments/generic/docker_service_kill/README.md:10]: Can we change this file? kubelet -> docker

File: chaoslib/litmus/docker_service_kill/docker_service_kill.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      wait_for:
+        timeout: "{{ c_duration }}"
+
+    - name: "[Status]: Checking the node status after chaos"
+      shell: |
+        kubectl get nodes {{ app_node }} --no-headers | awk '{print$2}'
+      args:
+        executable: /bin/bash
+      register: node_state
+      until: "'Ready' in node_state.stdout and 'Not' not in node_state.stdout"
```
Comment (ispeakc0de) [inline chaoslib/litmus/docker_service_kill/docker_service_kill.yml:97]: Can we directly use  `until: " 'NotReady' not in node_state.stdout"` ?

File: chaoslib/litmus/docker_service_kill/docker-service-kill.j2 [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    chaosUID: {{ chaos_uid }}
+{% endif %}        
+spec:
+  nodeName: {{ node_name }}
+  restartPolicy: Never
+  containers:
+  - name: service-kill
+    image: ubuntu:16.04
+    command: ["/bin/bash"]
+    args: ["-c", "sleep 10 && echo injecting_docker_crash.. && systemctl stop docker && sleep {{ c_duration }} && systemctl start docker && echo reverted_docker_crash.."]
```
Comment (ispeakc0de) [inline chaoslib/litmus/docker_service_kill/docker-service-kill.j2:19]: would you like to echo/print a conventional message(`echo "injecting docker crash..."`), or underscores are placed intentionally?

Comment (ghoshankur1983) [reply chaoslib/litmus/docker_service_kill/docker-service-kill.j2:19]: ok conventional messaging seems to be right idea.

**Comment by ghoshankur1983** [comment]
Please check the changes incorporated to accommodate the comments. Please close if everything is ok.

File: experiments/generic/docker_service_kill/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+<table>
+<tr>
+<th> Name </th>
+<th> Description </th>
+<th> Documentation Link </th>
+</tr>
+<tr>
+ <td> Docker Service Kill </td>
+ <td> This experiment causes kubelet service kill gracefully for a certain chaos duration. The experiment aims to verify resiliency of applications whose replicas may be evicted or becomes unreachable on account on nodes turning unschedulable (Not Ready) due to kubelet service kill. </td>
```
Comment (uditgaurav) [inline experiments/generic/docker_service_kill/README.md:11]: Still, some kubelet keywords are there. Can we change them to docker?

**Comment by ghoshankur1983** [comment]
Added the new changes as per Udit's comments. Kindly check.

**Pull request merged**